libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Templates/ActivityTemplate.cs

```diff
@@ -34,36 +35,11 @@ public virtual async Task<Activity> BindToData(ITurnContext context, object data
             if (!string.IsNullOrEmpty(this.Template))
             {
                 // if there is a message generator use that
-                IActivityGenerator activityGenerator = context.TurnState.Get<IActivityGenerator>();
-                if (activityGenerator != null)
-                {
-                    var result = await activityGenerator.Generate(
-                        turnContext: context,
-                        template: this.Template,
-                        data: data).ConfigureAwait(false);
-                    return result;
-                }
-
-                // fallback to just text based LG if there is a language generator
-                var message = Activity.CreateMessageActivity();
-                message.Text = this.Template;
-                message.Speak = this.Template;
-
-                ILanguageGenerator languageGenerator = context.TurnState.Get<ILanguageGenerator>();
-                if (languageGenerator != null)
-                {
-                    var result = await languageGenerator.Generate(
-                        turnContext: context,
-                        template: Template,
-                        data: data).ConfigureAwait(false);
-                    if (result != null)
-                    {
-                        message.Text = result;
-                        message.Speak = result;
-                    }
-                }
-
-                return message as Activity;
+                var result = await ActivityFactory.Generate(
```

Comment:
No, in here, you should call

ActivityFactory.CreateActivity(ILanguageGenerator.Generate(...)).